[AOTI] Add torch._export.aot_load

desertfire opened the pull request:
Summary: Add a torch._export.aot_load API that can load an AOTInductor-compiled model.so into a python executable.

Test Plan: CI

Differential Revision: D52825456




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

angelayi reviewed (approved):
nice! makes it a lot cleaner to run the executable now!

chenyang78 reviewed (approved):
LGTM. Thanks!

desertfire committed:
[AOTI] Add torch._export.aot_load (#117610)

Summary:
X-link: https://github.com/pytorch/benchmark/pull/2119


Add a torch._export.aot_load API that can load an AOTInductor-compiled model.so into a python executable.

Test Plan: CI

Reviewed By: khabinov, angelayi

Differential Revision: D52825456
